server/server.go

```diff
@@ -203,9 +207,15 @@ func (server *Server) setupServerEntryPoint(newServerEntryPointName string, newS
 	if metrics != nil {
 		serverMiddlewares = append(serverMiddlewares, middlewares.NewMetricsWrapper(metrics))
 	}
-	if server.globalConfiguration.Web != nil && server.globalConfiguration.Web.Statistics != nil {
-		statsRecorder = middlewares.NewStatsRecorder(server.globalConfiguration.Web.Statistics.RecentErrors)
-		serverMiddlewares = append(serverMiddlewares, statsRecorder)
+	if server.globalConfiguration.Web != nil {
+		metricsMiddleware := thoas_stats.New()
+		server.globalConfiguration.Web.MetricsRecorder = metricsMiddleware
+		serverMiddlewares = append(serverMiddlewares, metricsMiddleware)
+		if server.globalConfiguration.Web.Statistics != nil {
```

Comment:
@Juliens What is your mind about the deletion of the `metricsMiddleware` variable? 
```go
if server.globalConfiguration.Web.Statistics != nil {
    server.globalConfiguration.Web.MetricsRecorder = thoas_stats.New()
    serverMiddlewares = append(serverMiddlewares, server.globalConfiguration.Web.MetricsRecorder)
    }
```